### PR TITLE
feat(kumactl) check for incompatible versions on kumactl init

### LINKED ISSUE #725: Check if the kumactl version is the same as CP
### Summary

To avoid situation that a user by accident use old version of `kumactl` with newer Kuma CP we can check if versions of kumactl and CP are the same.

To do this, at the start of kumactl we can do the request to API Server on path `/` that return
```
{
  "tagline": "Kuma"
  "version": "0.4.0"
}
```
and see if the version is the same as in kumactl. If it is not, we can print a warning
```
Warning: Your kumactl version is 0.3.2 which is not the same as 0.4.0 for CP. Update your kumactl.
```
and execute commands regardless if it's in sync or not.

@subnetmarco thoughts?

### PR DESCRIPTION
**Pull request by tharun208**
### Summary
 Check kumactl version is in sync with kuma-cp
SUMMARY_GOES_HERE

### Full changelog
* Added apiServerClient
* Exposed apiServerClient through rootContext* 
* Added tests
### Issues resolved

Fix #725

### CODE REVIEW

File: app/kumactl/cmd/root.go [hunk 1]
```diff
@@ -19,6 +19,11 @@ import (
 	"github.com/Kong/kuma/pkg/cmd/version"
 	"github.com/Kong/kuma/pkg/core"
 	kuma_log "github.com/Kong/kuma/pkg/log"
+	kuma_version "github.com/Kong/kuma/pkg/version"
+)
+
+var (
+	kumactlLog = core.Log.WithName("kumactl")
```
Comment (nickolaev) [inline app/kumactl/cmd/root.go:26]: Hmm, do we want that? Shall we just go with the way other commands log their errors?

File: app/kumactl/cmd/root.go [hunk 2]
```diff
@@ -40,7 +49,9 @@ func NewRootCmd(root *kumactl_cmd.RootContext) *cobra.Command {
 			// once command line flags have been parsed,
 			// avoid printing usage instructions
 			cmd.SilenceUsage = true
-
+			if kumaCpInfo.Version != "" && kumaCpInfo.Version != kuma_version.Build.Version {
+				kumactlLog.Info("Warning: Your kumactl version is " + kuma_version.Build.Version + " which is not the same as " + kumaCpInfo.Version + " for CP. Update your kumactl.")
```
Comment (nickolaev) [inline app/kumactl/cmd/root.go:53]: Maybe not Info? Warning if possible.

File: app/kumactl/pkg/resources/apiserver_client_test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+				}),
+			},
+		}
+
+		// when
+		_, err := client.GetVersion()
+
+		// then
+		Expect(err).To(MatchError("unexpected status code 400"))
+	})
```
Comment (nickolaev) [inline app/kumactl/pkg/resources/apiserver_client_test.go:31]: Shuould you also test other scenarios? Maybe success, various versions etc.

File: app/kumactl/pkg/cmd/root_context.go [hunk 4]
```diff
@@ -241,3 +244,15 @@ func (rc *RootContext) cpOnTheSameMachine() (bool, error) {
...
+func (rc *RootContext) Version() (types.IndexResponse, error) {
+	controlPlane, err := rc.CurrentControlPlane()
+	if err != nil {
+		return types.IndexResponse{}, err
+	}
+	client, err := rc.Runtime.NewAPIServerClient(controlPlane.Coordinates.ApiServer.Url)
+	if err != nil {
+		return types.IndexResponse{}, errors.Wrap(err, "could not create components client")
+	}
+	return client.GetVersion()
```
Comment (nickolaev) [inline app/kumactl/pkg/cmd/root_context.go:257]: I somehow feel there shoudl be closer call for the `client` but it is not available. This way you leak resources IMO.

**Comment by nickolaev** [comment]
@tharun208 would you consider addressing the comments here?

**Comment by tharun208** [comment]
Will add the tests for checking the version mismatch text @nickolaev

**Pull request merged**